cpp/src/io/parquet/chunk_dict.cu

```diff
@@ -0,0 +1,372 @@
+  __shared__ size_type s_ck_start_val_idx;
+  __shared__ size_type s_num_values;
+
+  if (t == 0) {
+    // Find the chunk this block is a part of
+    size_type num_rowgroups = chunks.size().first;
+    size_type rg_idx        = 0;
+    while (rg_idx < num_rowgroups) {
+      if (auto ck = chunks[rg_idx][col_idx];
+          start_row >= ck.start_row and start_row < ck.start_row + ck.num_rows) {
+        break;
+      }
+      ++rg_idx;
+    }
+    s_chunk = chunks[rg_idx][col_idx];
+    s_col   = *(s_chunk.col_desc);
+
+    // Find the bounds of values in leaf column to be inserted into the map for current chunk
+
+    auto col                 = *(s_col.parent_column);
+    auto start_value_idx     = start_row;
+    auto end_value_idx       = end_row;
+    auto chunk_start_val_idx = s_chunk.start_row;
+    while (col.type().id() == type_id::LIST or col.type().id() == type_id::STRUCT) {
+      if (col.type().id() == type_id::STRUCT) {
+        start_value_idx += col.offset();
+        chunk_start_val_idx += col.offset();
+        end_value_idx += col.offset();
+        col = col.child(0);
+      } else {
+        auto offset_col     = col.child(lists_column_view::offsets_column_index);
+        start_value_idx     = offset_col.element<size_type>(start_value_idx + col.offset());
+        chunk_start_val_idx = offset_col.element<size_type>(chunk_start_val_idx + col.offset());
+        end_value_idx       = offset_col.element<size_type>(end_value_idx + col.offset());
+        col                 = col.child(lists_column_view::child_column_index);
+      }
+    }
+    s_start_value_idx  = start_value_idx;
+    s_ck_start_val_idx = chunk_start_val_idx;
+    s_num_values       = end_value_idx - start_value_idx;
+  }
+  __syncthreads();
+
+  if (not s_chunk.use_dictionary) { return; }
+
+  column_device_view const& data_col = *s_col.leaf_column;
+
+  auto map = map_type::device_view(
+    s_chunk.dict_map_slots, s_chunk.dict_map_size, KEY_SENTINEL, VALUE_SENTINEL);
+
+  for (size_t i = 0; i < s_num_values; i += block_size) {
+    if (t + i < s_num_values) {
+      auto val_idx = s_start_value_idx + t + i;
+      bool is_valid =
+        (i + t < s_num_values && val_idx < data_col.size()) ? data_col.is_valid(val_idx) : false;
+
+      if (is_valid) {
+        auto found_slot = type_dispatcher(data_col.type(), map_find_fn{map}, data_col, val_idx);
+        cudf_assert(found_slot != map.end() &&
+                    "Unable to find value in map in dictionary index construction");
+        if (found_slot != map.end()) {
+          // No need for atomic as this is not going to be modified by any other thread
+          auto* val_ptr = reinterpret_cast<map_type::mapped_type*>(&found_slot->second);
+          s_chunk.dict_index[val_idx - s_ck_start_val_idx] = *val_ptr;
+        }
+      }
+    }
+  }
+}
+
+void InitializeChunkHashMaps(device_span<EncColumnChunk> chunks, rmm::cuda_stream_view stream)
+{
+  constexpr int block_size = 1024;
```

Comment:
Use `int32_t` instead, for clarification.

Reply:
I believe kernel launch parameters are not sensitive to bit size.